src/rubrix/client/datasets.py

```diff
@@ -153,88 +155,146 @@ def _to_datasets_dict(self) -> Dict:
+        }
 
-        dataset = cls._prepare_hf_dataset(
+        cols_to_be_renamed, cols_to_be_joined = {}, {}
+        for field, col in column_mapping.items():
+            if field in cls._RECORD_FIELDS_WITH_MULTIPLE_INPUT_COLUMNS:
+                cols_to_be_joined[field] = [col] if isinstance(col, str) else col
+            else:
+                cols_to_be_renamed[col] = field
+
+        dataset = dataset.rename_columns(cols_to_be_renamed)
+
+        dataset = cls._remove_unsupported_columns(
             dataset,
-            id=id,
-            text=text,
-            annotation=annotation,
-            metadata=metadata,
-            **kwargs,
+            extra_columns=[col for cols in cols_to_be_joined.values() for col in cols],
         )
 
+        return dataset, cols_to_be_joined
+
+    @classmethod
+    def _remove_unsupported_columns(
+        cls,
+        dataset: "datasets.Dataset",
+        extra_columns: List[str],
+    ) -> "datasets.Dataset":
+        """Helper function to remove unsupported columns from the `datasets.Dataset` following the record type.
+
+        Args:
+            dataset: The dataset.
+            extra_columns: Extra columns to be kept.
+
+        Returns:
+            The dataset with unsupported columns removed.
+        """
         not_supported_columns = [
-            col for col in dataset.column_names if col not in cls._record_init_args()
+            col
+            for col in dataset.column_names
+            if col not in cls._record_init_args() + extra_columns
         ]
+
         if not_supported_columns:
             _LOGGER.warning(
                 f"Following columns are not supported by the {cls._RECORD_TYPE.__name__}"
                 f" model and are ignored: {not_supported_columns}"
             )
             dataset = dataset.remove_columns(not_supported_columns)
-        return cls._from_datasets(dataset)
 
-    @classmethod
-    def _prepare_hf_dataset(
-        cls,
-        dataset: "dataset.Dataset",
-        id: Optional[str] = None,
-        text: Optional[str] = None,
-        annotation: Optional[str] = None,
-        metadata: Optional[Union[str, List[str]]] = None,
-    ) -> "dataclasses.Dataset":
-        for field, parser in [
-            (id, cls._parse_id_field),
-            (text, cls._parse_text_field),
-            (metadata, cls._parse_metadata_field),
-            (annotation, cls._parse_annotation_field),
-        ]:
-            if field:
-                dataset = parser(dataset, field)
         return dataset
 
-    @classmethod
-    def _from_datasets(cls, dataset: "datasets.Dataset") -> "Dataset":
-        """Helper method to create a Rubrix Dataset from a datasets Dataset.
+    @staticmethod
+    def _join_datasets_columns_and_delete(
+        row: Dict[str, Any], columns: List[str]
+    ) -> Dict[str, Any]:
+        """Joins columns of a `datasets.Dataset` row into a dict, and deletes the single columns.
+        UPDATES TE ROW!
```

Comment:
?

Reply:
Typo, wanted to explicitly state that the method updates the ``row`` dictionary.